### PR TITLE
Add -a rate_adj and -e dither options to dmk2cw.

### PR DESCRIPTION
**Pull request by TimothyPMann**
Plus: detect and error out if too high a -c value is used for the format.

### CODE REVIEW

File: dmk2cw.man [hunk 1]
```diff
@@ -229,6 +229,28 @@ disk intact.  With fill = 0x1nn, the fill pattern is 0xnn encoded
 in FM.  With fill = 0x2nn, the fill pattern is 0xnn encoded in
 MFM.
 .TP
+.B \-a \fIrate_adj\fP
+Causes dmk2cw to multiply the average data rate by rate_adj (default
+1.0).  Values should be close to 1.0; for example, 1.005 makes the
+data rate 0.5 percent faster, while 0.995 makes it 0.5% slower.
+Making the data rate 1% faster has a similar effect to running the
+drive motor %1 slower: more data fits on the track.  Of course if the
```
Comment (qbarnes) [inline dmk2cw.man:237]: In this paragraph, you refer to "0.5 percent faster", "0.5% slower" and "%1".  Would prefer a little more consistency.  When you get a chance in a later commit, please fix "0.5 percent" to just "0.5%" and "%1" to 1%".

File: dmk2cw.man [hunk 2]
```diff
@@ -229,6 +229,28 @@ disk intact.  With fill = 0x1nn, the fill pattern is 0xnn encoded
...
+the intervals between transitions by +/-1 when needed to keep the
+average data rate accurate.  If dither is 0 (the default), no such
+adjustment is done.  As a result, the interval sizes are consistent
+across the whole track, but the average data rate may be off
+(typically by much less than 1%) because of cumulative rounding error.
+If a disk needs the -a option to adjust the average data rate very
+precisely, turning on dither too may help.  Dither is a relatively new
+feature, and it is not yet clear whether it is an improvement idea or
+whether the occasional +/-1 tick will make disks more difficult to
+read.
```
Comment (qbarnes) [inline dmk2cw.man:252]: I think this feature is close or the same as a discussion I had with Jens back a few years back.  At one point he was thinking of changing the firmware to make the adjustment automatic.  I think I convinced him that the error was so small that it wouldn't ever really matter.  But it still seems like a worthwhile experimental feature to make available.

**Review by qbarnes — APPROVED** [review]

**Comment by TimothyPMann** [comment]
Ugh, I usually don't make mistakes like "%1". I'll fix that as well as using the symbol instead of the word throughout.
________________________________

**Comment by TimothyPMann** [comment]
Fixing the dmk2cw.man typo in my rate_adj branch and doing a new pull request doesn't look like it is going to do the right thing in github. github previews this pull request as merging the whole branch again and tells me the branch can't be automatically merged. Is the preview just bogus or should I do this some other way?

**Pull request merged**